sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -364,6 +364,15 @@ object DateTimeUtils {
     }
   }
 
+  def stringToTimestamp(s: UTF8String, timeZoneId: ZoneId, ansiEnabled: Boolean): Option[Long] = {
```

Comment:
It's better to add a `stringToTimestampAnsi` method. Non-ansi code path should call the original `stringToTimestamp` method.